src/main/java/catan/services/impl/PlayServiceImpl.java

```diff
@@ -115,27 +119,40 @@ private void buildCity(UserBean user, GameBean game, String nodeId) throws PlayE
     }
 
     private void endTurn(GameBean game) throws GameException {
-        boolean shouldUpdateNextMove = true;
-
-        if (game.getStage().equals(GameStage.PREPARATION)) {
-            Integer previousPreparationCycle = game.getPreparationCycle();
-            preparationStageUtil.updateGameStageToMain(game); //TODO: move it to the end of method calls
-            preparationStageUtil.updateCurrentCycleInitialBuildingNumber(game);
-            preparationStageUtil.updatePreparationCycle(game);
-            Integer newPreparationCycle = game.getPreparationCycle();
-            boolean preparationFinished = newPreparationCycle == null && !game.getCurrentMove().equals(1);
-            shouldUpdateNextMove = (preparationFinished || previousPreparationCycle.equals(newPreparationCycle));
-        }
+        switch (game.getStage()) {
+
+            case PREPARATION:
+                Integer previousPreparationCycle = game.getPreparationCycle();
+                preparationStageUtil.updateGameStageToMain(game); //TODO: move it to the end of method calls
+                preparationStageUtil.updateCurrentCycleInitialBuildingNumber(game);
+                preparationStageUtil.updatePreparationCycle(game);
+                Integer newPreparationCycle = game.getPreparationCycle();
+                boolean preparationFinished = newPreparationCycle == null && !game.getCurrentMove().equals(1);
+                if (preparationFinished || previousPreparationCycle.equals(newPreparationCycle)) {
+                    playUtil.updateNextMove(game);
```

Comment:
since in this case block we call only methods from preparationStageUtil, maybe we should avoid calling playUtil here?